### PR TITLE
Set trainer's device to CPU if it is initially loaded onto MPS

### PR DESCRIPTION
**Pull request by Lilferrit**

### CODE REVIEW

File: casanovo/denovo/model_runner.py [hunk 1]
```diff
@@ -400,6 +400,14 @@ def initialize_trainer(self, train: bool) -> None:
 
         self.trainer = pl.Trainer(**trainer_cfg)
 
+        # Reinitialize the trainer if it was loaded on to mps
+        if self.trainer.strategy.root_device.type == "mps":
```
Comment (bittremieux) [inline casanovo/denovo/model_runner.py:404]: **question:** Is the device available before the trainer is initialized? If so, I'd keep changing from MPS to CPU consistent with how the number of workers is set (i.e. utils function and modify it [directly during config loading](https://github.com/Noble-Lab/casanovo/blob/ec67e980019a721ec91debe1068692ba78e13ac3/casanovo/config.py#L154). That has as benefit that config settings are interpreted in a single location, and makes it easier to add more logic if other accelerators also have compatibility issues. (+ Small extra advantage because of not having to initialize the trainer twice.)

Comment (Lilferrit) [reply casanovo/denovo/model_runner.py:404]: This was unsatisfying to me too but alas it doesn't look like there is really a way to tell how pytorch lightning will resolve `accelerator="auto"` without initializing the trainer first, nor is there a way to move the trainer once it's initialized.

Another way to do this would be to just check if the machine is Apple silicon and then setting the accelerator to CPU if so, which could be done in the config object, however I feel like this would introduce an edge case if a user is on Apple Silicon but also has a GPU available.

**Comment by Lilferrit** [comment]
Sure thing, I just updated the changelog.

**Comment by Lilferrit** [comment]
I just tested the MGF file attatched to #285 on an m.3 Macbook and this fix does resolve the issue when using the default config.
